mvc/src/main/java/com/interface21/webmvc/servlet/mvc/DispatcherServlet.java

```diff
@@ -40,17 +45,30 @@ protected void service(final HttpServletRequest request, final HttpServletRespon
 
         try {
             final var controller = handlerMappingRegistry.getHandler(request);
-            final var modelAndView = handlerAdapterRegistry.getHandlerAdapter(controller)
+            final var Object = handlerAdapterRegistry.getHandlerAdapter(controller)
                 .handle(request, response, controller);
 
-            move(modelAndView, request, response);
+            move(Object, request, response);
         } catch (Throwable e) {
             log.error("Exception : {}", e.getMessage(), e);
             throw new ServletException(e.getMessage());
         }
     }
 
-    private void move(final ModelAndView view, final HttpServletRequest request, final HttpServletResponse response) throws Exception {
-        view.getView().render(view.getModel(), request, response);
+    private void move(final Object object, final HttpServletRequest request, final HttpServletResponse response) throws Exception {
+        ModelAndView modelAndView = null;
+        if (object instanceof String) {
+            modelAndView = new ModelAndView(viewResolver.resolveViewName((String) object));
+        }
+        if (object instanceof ModelAndView) {
+            modelAndView = (ModelAndView) object;
+            modelAndView.getView().render(modelAndView.getModel(), request, response);
+        }
+
+        if (modelAndView == null) {
+            throw new IllegalStateException("지원되지 않는 핸들러입니다.");
+        }
+
+        modelAndView.getView().render(modelAndView.getModel(), request, response);
```

Comment:
컨트롤러나 어댑터는 getHandler, getHandlerAdapter인데, viewResolving은 왜 직접  if문으로 동작을 하는걸까요! 

```suggestion
        ViewResolver resolver = viewResolvers.findResolver(object, request, response).orElseThrow(...);
        resolver.resolve(object, request, response);
```

좀 더 추상화시켜볼 수 있지 않을까요!